ide-backend/IdeSession/Update.hs

```diff
@@ -179,15 +179,21 @@ initSession initParams@SessionInitParams{..} ideConfig@SessionConfig{..} = do
   verifyConfig ideConfig
 
   configDirCanon <- Dir.canonicalizePath configDir
-  ideSessionDir  <- createTempDirectory configDirCanon "session."
-  let ideStaticInfo = IdeStaticInfo{..}
+  ideSessionDir' <- createTempDirectory configDirCanon "session."
+  ideSessionDir <- case configLocalWorkingDir of
```

Comment:
Why is ideSessionDir being set to the local working directory?  I think the session directory stuff should be left the same, and configLocalWorkingDir should only matter for where project files are stored.  Doing this will fix the issue where rpc folders are placed in the project directory.